roles/builder/tasks/main.yml

```diff
@@ -15,26 +15,78 @@
   osbuild.composer.push_blueprint:
     src: "{{ builder_blueprint_src_path }}"
 
+- name: Check if blueprint directory exists
+  ansible.builtin.stat:
+    path: "/var/www/html/{{ builder_blueprint_name }}"
+  register: stat_output
+
+- name: Initialize rpm-ostree repo for blueprint
+  when: not stat_output.stat.exists
+  block:
+    - name: Create blueprint directory
+      ansible.builtin.file:
+        path: "/var/www/html/{{ builder_blueprint_name }}"
+        mode: 0755
+        state: directory
+    - name: Initialize repository
+      ansible.builtin.command:
+        chdir: "/var/www/html/{{ builder_blueprint_name }}"
+        cmd: '/usr/bin/ostree --repo=repo init --mode=archive'
+      args:
+        creates: "/var/www/html/{{ builder_blueprint_name }}/repo"
+
+
 - name: Start compose
   osbuild.composer.start_compose:
-    blueprint: "my-rhel-edge-test"
+    blueprint: "{{ builder_blueprint_name }}"
     compose_type: "{{ builder_compose_type }}"
   register: compose_start_out
 
 - name: Wait for compose to finish
   osbuild.composer.wait_compose:
     compose_id: "{{ compose_start_out['result']['build_id'] }}"
 
+- name: Create tmp directory for blueprint
+  ansible.builtin.file:
+    path: "/tmp/{{ builder_blueprint_name }}"
+    mode: 0755
+    state: directory
+
 - name: Export the compose artifact
   osbuild.composer.export_compose:
     compose_id: "{{ compose_start_out['result']['build_id'] }}"
-    dest: "/tmp/composer-compose.{{ compose_start_out['result']['output_type'] }}"
+    dest: "/tmp/{{ builder_blueprint_name }}/{{ builder_blueprint_name }}-{{ blueprint_output['current_version'] }}.{{ compose_start_out['result']['output_type'] }}" # noqa yaml[line-length]
+
+- name: Update ostree repository
+  when: compose_start_out['result']['output_type'] == 'tar'
+  block:
+    - name: Untar artifact
+      ansible.builtin.unarchive:
+        src: /tmp/{{ builder_blueprint_name }}/{{ builder_blueprint_name }}-{{ blueprint_output['current_version'] }}.tar
+        dest: /tmp/{{ builder_blueprint_name }}
+        remote_src: true
+
+    - name: Get checksum from artifact
+      ansible.builtin.command:
+        cmd: "/usr/bin/ostree --repo=/tmp/{{ builder_blueprint_name }}/repo rev-parse {{ builder_blueprint_ref }}"
+      register: checksum_output
+      changed_when: false
+
+    - name: Pull commit from artifact
+      ansible.builtin.command:
+        cmd: "/usr/bin/ostree --repo=/var/www/html/{{ builder_blueprint_name }}/repo pull-local /tmp/{{ builder_blueprint_name }}/repo {{ checksum_output['stdout'] }}"
+      changed_when: true
+
+    - name: Commit changes to repository
+      ansible.builtin.command:
+        cmd: "/usr/bin/ostree --repo=/var/www/html/{{ builder_blueprint_name }}/repo commit -b {{ builder_blueprint_ref }} -s 'Release {{ blueprint_output['current_version'] }}' --add-metadata-string=version={{ blueprint_output['current_version'] }} --tree=ref={{ checksum_output['stdout'] }}"
+      changed_when: true
```

Comment:
Let's add tasks here to delete the `composer-composer-*.tar` file we created above and the directory of the expanded tarball so we're not just filling up space on the device. This is expecially important for Fedora where `/tmp/` is a `tmpfs` by default and therefore resides in RAM because `tmpfs` is a ramdisk type.

Reply:
Sounds good to me! However aren't we testing to see if that tar files exists in https://github.com/ansible-collections/osbuild.composer/blob/main/tests/integration/targets/role_builder/tasks/tests.yml

Reply:
That was originally written when the export of the build payload was a separate task/module and the role left it on disk. If we change the behavior, we should change that test case :)